platform/state/platform_state.proto

```diff
@@ -0,0 +1,391 @@
+     * All existing node identifiers SHALL be strictly less than this value.<br/>
+     */
+    NodeId next_node_id = 2;
+
+    /**
+     * A list of all consensus node addresses.
+     * <p>
+     * This list SHALL NOT be empty.
+     * If a consensus node is not in this list it SHALL NOT participate in the
+     * network.
+     */
+    repeated Address addresses = 3;
+}
+
+/**
+ * A single network address.
+ * This is one address in the network address book, including all required
+ * information to include that consensus node in the consensus gossip.
+ *
+ * ### Implementation Detail
+ * When `enableEventStreaming` is set to true, the memo field is REQUIRED
+ * and MUST be unique.
+ */
+message Address {
+    /**
+     * The ID of this member.
+     * <p>
+     * This identifier SHALL be agreed upon by all consensus nodes.
+     */
+    NodeId id = 1;
+
+    /**
+     * The nickname this consensus node uses to refer to another consensus node.
+     */
+    string nickname = 2;
+
+    /**
+     * The name that a consensus node uses to refer to themselves.
+     */
+    string self_name = 3;
+
+    /**
+     * A consensus weight.<br/>
+     * This value is the relative weight for this consensus node in the
+     * consensus voting algorithm.
+     */
+    uint64 weight = 4;
+
+    /**
+     * A string network address.<br/>
+     * This value is the hostname assigned on the "internal" network
+     * interface behind any network firewalls, protocol gateways, or
+     * DNS translations.
+     * <p>
+     * This value SHALL be either a FQDN host name or an IPv4 address.
+     */
+    string hostname_internal = 5;
+
+    /**
+     * Network port number.<br/>
+     * This value is the "translated" port number assigned on the "internal"
+     * network behind any network firewalls or other address translations.
+     * <p>
+     * This value SHALL be between 1 and 65535.
+     */
+    uint32 port_internal = 6;
+
+    /**
+     * A string network address.<br/>
+     * This value is the hostname assigned on the "public" network
+     * interface visible to the general internet.
+     * <p>
+     * This value SHALL be either a FQDN host name or an IPv4 address.
+     */
+    string hostname_external = 7;
+
+    /**
+     * Network port number.<br/>
+     * This value is the port number visible to the general internet.
+     * <p>
+     * This value SHALL be between 1 and 65535.
+     */
+    uint32 port_external = 8;
+
+    /**
+     * The signing x509 certificate of the consensus node.
+     * <p>
+     * This SHALL provide the public key used for signing.
+     */
+    bytes sig_cert = 9;
+
+    /**
+     * The agreement x509 certificate of the consensus node.
+     * <p>
+     * This SHALL be used for establishing TLS connections.
+     */
+    bytes agree_cert = 10;
+
+    /**
+     * A string that provides additional information about this consensus node.
+     */
+    string memo = 11;
```

Comment:
This also probably doesn't belong in the platform data structure in the long term. If we can't get rid of it in the short term, we should move it to a high index.

Reply:
agreed.

Reply:
The field is updated with a high index.